modApiServer/src/org/aion/api/server/nrgprice/NrgBlockPriceStrategy.java

```diff
@@ -0,0 +1,139 @@
+package org.aion.api.server.nrgprice;
+
+import org.aion.base.type.Address;
+import org.aion.log.AionLoggerFactory;
+import org.aion.log.LogEnum;
+import org.aion.zero.impl.types.AionBlock;
+import org.aion.zero.types.AionTransaction;
+import org.slf4j.Logger;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+
+/* Implementation of strategy adopted by Ethereum mainstream clients in early 2018 of using the
+ * notion of 'blockPrice': minimum nrgPriced transaction in a block that did not originate from the proposer
+ * of that block. This algorithm recommends the Nth percentile blockPrice observed over the last M blocks
+ *
+ * This design has a bias toward the lower end of the threshold of nrg prices required
+ * for txns to be accepted into blocks (heuristically)
+ *
+ * A transaction based design alternatively has been observed in the wild to have a positive feedback
+ * effect where large numbers of people following the recommendation will tend the recommendation upward
+ *
+ */
+public class NrgBlockPriceStrategy extends NrgPriceAdvisor<AionBlock, AionTransaction> {
+
+    protected static final Logger LOG = AionLoggerFactory.getLogger(LogEnum.API.name());
+    private ArrayBlockingQueue<Long> blkPriceQ;
+
+    // enforce writes through buildRecommendation() to happen serially if multiple objects try to hold reference
+    // to this object. ie. enforce only one writer to populate the block list at a time
+    // private Object recommendationBuilderLock = new Object();
+
+    int percentile;
+    int windowSize;
+    int recommendationIndex;
+
+    public NrgBlockPriceStrategy(long defaultPrice, long maxPrice, int windowSize, int percentile) {
+        super(defaultPrice, maxPrice);
+
+        // clamp the percentile measure
+        if (percentile < 0)
+            this.percentile = 0; // pick the smallest value
+        else if (percentile > 100)
+            this.percentile = 100; // pick the largest value
+        else
+            this.percentile = percentile;
+
+        // clamp the windowSize at the bottom
+        if (windowSize < 0)
+            this.windowSize = 0; // when no elements in window, just return the minPrice
```

Comment:
Clamping windowSize to be 0 causes ``processBlock(AionBlock blk)`` to log and error. Suggestion is to add an extra case to ``processBlock`` to detect when windowSize is 0 and simply return.

Reply:
workaround for this: restricted minimum window size to 1 in constructor.